### ISSUE TITLE
Document store broke in 1.3.0?

### ISSUE THREAD

**Issue by timothytierney** [tariqdaouda/pyArango#68]
I am fetching a document from arangodb3 (v3.0.12) using pyArango 1.3.0..  The doc type returned is <class 'pyArango.document.DocumentStore'> which appear right.  I then make an assignment in and my JSON appears to have all these <store:xxx> embedded keys which breaks the existing code.  1.2.8 worked fine.  Is this the intended effect?

Here is the output:

doc = threadCollection.fetchDocument(threadname)
threadJSON = doc["config"]

threadJSON now reads:
<store: {'thread': <store: {'_comment': 'p9s2 system for a min 2 host configuration for xxxx development', 'networks': [{'networktype': 'management', 'networkNamespace': 'global'}, {'networktype': 'data1', 'networkNamespace': 'global'}, {'networktype': 'data2', 'networkNamespace': 'global'}, {'networktype': 'ipmi', 'networkNamespace': 'global'}], 'name': 'p9s2', 'hostlist': ['p9s2_vmhost1', 'p9s2_xxxx', 'p9s2_xxxx'], 'application_config': <store: {'version': 'LATEST', 'docker': <store: {'registry': <store: {'ip': 'xx.xxx.xx.xx', 'port': 'xxxx'}>}>, 'applicationtype': 'xxxx', 'kub_network': <store: {'pod_network_cidr': 'xxx.xx.x.x/xx'}>}>, 'guestlist': ['p9s2_xxxx', 'p9s2_xxx', 'p9s2_xxx', 'p9s2_xxx', 'p9s2xxxx'], 'puppetcontrol': <store: {'host': 'master', 'guest': 'master'}>, 'masteragent': <store: {'ip': 'xx.xxx.xxx.x', 'username': 'cobbler', 'password': 'cobbler'}>, 'orchestrationnode': <store: {'enabled': False}>, 'cobblercontrol': <store: {'host': 'master', 'guest': 'master'}>, 'slaveagent': <store: {'ip': 'xxx.xxx.x.x', 'username': 'cobbler', 'password': 'cobbler'}>}>}>

v1.2.8 threadJSON was (using same code)
{'thread': {'orchestrationnode': {'enabled': False}, '_comment': 'p9s2 system for a min 2 host configuration for xxx development', 'name': 'p9s2', 'guestlist': ['p9s2_xxxx', 'p9s2_xxxx', 'p9s2_xxxx', 'p9s2_xxxx', 'p9s2_xxxx'], 'hostlist': ['p9s2_xxxx', 'p9s2_xxxx', 'p9s2_xxxx'], 'puppetcontrol': {'guest': 'master', 'host': 'master'}, 'masteragent': {'password': 'cobbler', 'ip': 'xx.xxx.xxx.x', 'username': 'cobbler'}, 'application_config': {'applicationtype': 'xxxx', 'version': 'LATEST', 'docker': {'registry': {'ip': 'xx.xxx.xxx.xx', 'port': 'xxxx'}}, 'kub_network': {'pod_network_cidr': 'xxx.xx.x.x/xx'}}, 'networks': [{'networkNamespace': 'global', 'networktype': 'management'}, {'networkNamespace': 'global', 'networktype': 'data1'}, {'networkNamespace': 'global', 'networktype': 'data2'}, {'networkNamespace': 'global', 'networktype': 'ipmi'}], 'cobblercontrol': {'guest': 'master', 'host': 'master'}, 'slaveagent': {'password': 'cobbler', 'ip': 'xxx.xxx.x.x', 'username': 'cobbler'}}}

**Comment by tariqdaouda** [comment]
Yes, this an expected behaviors, DocumentStore have replaced dicts. However they should be usable transparently, exactly the same way as dicts. Can you tell me exactly how your existing code was broken?

If you need a pure dict representation of your object you can use:
`threadJSON._store.getStore()` and  `threadJSON._store.getPatches()`, if you only need the modifications since the last save.

**Comment by timothytierney** [comment]
doc = threadCollection.fetchDocument(threadname)
threadJSON = doc["config"] (this would give me a valid json dict)

The threadJSON was expecting to be a dict so it couldn't find any of the keys because of the embedded <store:xxx>.  I'll try your suggestions.  I was expecting a return of the 1.2.8 {'thread': { threadJSON that I listed but instead it returned the <store: {'thread': <store: 

I am a little worried that future changes are not backward compatible and will require changes to our code which could break production code.

Thank you for the quick response.

**Comment by timothytierney** [comment]
Also we are seeing an issue in the released code (1.3.0) that does not include the getStore procedure as is in gitHub.  The document class installed from the 1.3.0 version != to git hub.  Code looks ok on github, but that is not what is in the installed package.

**Comment by tariqdaouda** [comment]
Don't worry about the backward compatibility. It is very important to me as well. `DocumentStore` should not break it they are just a better way to deal with nested documents.

I have just downloaded the release code of 1.3.0 on GitHub and the PyPI archive and both have `DocumentStore` and `getStore()`. What issue are you referring to?

**Comment by timothytierney** [comment]
A "pip install pyarango" gave us v1.3.0.  That installation is different than what is on github.  Is pip install pyarango not a valid why to install the latest package?  Do I have to go directly to github?

Are you also saying the json should be valid with the <store:xx> tags?

**Comment by tariqdaouda** [comment]
The upgrade command would have been: `pip install --upgrade pyArango`
Yes, the json should always be valid. It is automatically generated inside the object and has never been accessible from outside the objects.

**Comment by timothytierney** [comment]
The "pip install pyarango" was done on a new installation, not an upgrade.
